providers/implementations/capabilities/tls.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/crypto.h>
+#include <openssl/core_numbers.h>
+#include <openssl/core_names.h>
+#include <openssl/params.h>
+#include "prov/implementations.h"
+#include "prov/provider_ctx.h"
+
+static OSSL_OP_capabilities_supported_fn tls_group_supported;
+
+typedef struct tls_group_info_st {
+    int id;                /* Group ID */
+    const char *name;           /* Group Name */
+    const char *algorithmm;     /* Name of algorithm */
+    int secbits;                /* Bits of security (from SP800-57) */
+} TLS_GROUP_INFO;
+
+static const TLS_GROUP_INFO group_list[] = {
+    { 0x0001, "sect163k1", "EC", 80 },
+    { 0x0002, "sect163r1", "EC", 80 },
+    { 0x0003, "sect163r2", "EC", 80 },
+    { 0x0004, "sect193r1", "EC", 80 },
+    { 0x0005, "sect193r2", "EC", 80 },
+    { 0x0006, "sect233k1", "EC", 112 },
+    { 0x0007, "sect233r1", "EC", 112 },
+    { 0x0008, "sect239k1", "EC", 112 },
+    { 0x0009, "sect283k1", "EC", 128 },
+    { 0x000A, "sect283r1", "EC", 128 },
+    { 0x000B, "sect409k1", "EC", 192 },
+    { 0x000C, "sect409r1", "EC", 192 },
+    { 0x000D, "sect571k1", "EC", 256 },
+    { 0x000E, "sect571r1", "EC", 256 },
+    { 0x000F, "secp160k1", "EC", 80 },
+    { 0x0010, "secp160r1", "EC", 80 },
+    { 0x0011, "secp160r2", "EC", 80 },
+    { 0x0012, "secp192k1", "EC", 80 },
+    { 0x0013, "secp192r1", "EC", 80 },
+    { 0x0014, "secp224k1", "EC", 112 },
+    { 0x0015, "secp224r1", "EC", 112 },
+    { 0x0016, "secp256k1", "EC", 128 },
+    { 0x0017, "secp256r1", "EC", 128 },
+    { 0x0018, "secp384r1", "EC", 192 },
+    { 0x0019, "secp521r1", "EC", 256 },
+    { 0x001A, "brainpoolP256r1", "EC", 128 },
+    { 0x001B, "brainpoolP384r1", "EC", 192 },
+    { 0x001C, "brainpoolP512r1", "EC", 256 },
+    { 0x001D, "x25519", "X25519", 128 },
+    { 0x001E, "x448", "X448", 224 },
+    /* Security bit values for FFDHE groups are as per RFC 7919 */
+    { 0x0100, "ffdhe2048", "DH", 103, },
```

Comment:
Maybe we should stick to the values returned by BN_security_bits() for these instead.. i.e 2048 = 112 bits etc.
i.e if we were to ever enforce these values (which we probably wont) then ffdhe would be rejected in fips mode (if 112 bits is required).

Reply:
@paulidale  ??

Reply:
+1

Reply:
I changed it to the BN_security_bits() values as suggested.